articles/why-node-js-is-popular/index.md

```diff
@@ -0,0 +1,90 @@
+Among many developers, JavaScript skills are mandatory. Node.js uses JavaScript. So, learning Node.js is relatively easy and few weeks of learning can get you up and running. Without JavaScript Knowledge, learning Node.js will take a bit longer. Therefore, learning Node.js is not that tricky.
+
+#### 2. The scalability offered.
+Node.js is designed with scalable network applications in mind. Node.js can handle many concurrent requests. This is the main reason it quickly became popular among developers and large companies. With the ability to handle many simultaneous requests, the server does not strain. A callback is fired only after a request, but if there are no more jobs to be done, Node.js will go to sleep. Node.js is only online when needed. This avoids unnecessary usage of your RAM.
+
+### 3. The Mobile-friendly, Cross-platform, and Dev-Friendly nature of Node.js.
+The ability to build flexible apps that run smoothly on any platform, be it Linux, Windows, or Mac OS gives Node.js a big thumbs up. There has been a rise in cross-platform app development in the last few years. The fear of apps not running on other operating systems is gone, as Node.js helped take care of that. It has enabled developers to write apps that run on different platforms with little or no modifications being made.
+
+#### 4. Node.js is light and fast.
+Node.js uses Google's V8 JavaScript engine which is built in C++. V8 engine handles memory allocation, compiles, and executes JavaScript with lightning speed. More about Google's V8 JS engine can be found in the V8 [docs](https://v8.dev/docs). Node.js uses a non-blocking model in performing operations. Node.js has published a full article on blocking and non-blocking models [here](https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/). It handles requests made in a single asynchronous thread. This reduces CPU workloads as well as memory. This makes your app lightweight.
+
+#### 5.The many hosting providers available.
+As a result of its growth in recent years, Many cloud-based hosting providers and web servers have added quite simple ways of hosting Node.js code. Some of these hosting providers include Heroku, Amazon Web Services, and DigitalOcean among others. There are also many detailed and simple guides on this.
+
+#### 6. Highly extensible
+Node.js does not provide everything that you need all at once. Instead, you customize or add what you need over time. This prevents having unnecessary libraries. [Npm](https://www.npmjs.com/) - **Node package manager** is the default package manager in Node.js. Here, you can browse and install over 1 million open source packages. You can also use it to identify and install project dependencies.
+
+#### 7. Its caching ability
+Node.js provides the ability to cache single modules. Caching is the storage of data for future requests. When a request is made, resources are fetched from the cache instead of the main servers. Thus, the web pages load faster and responds quickly.
```

Comment:
The sentence starts with caching of modules and then moves on to caching of pages? I am assuming we want to talk about caching modules here?